plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/FileSystemSpoolingModule.java

```diff
@@ -78,6 +82,19 @@ protected void setup(Binder binder)
 
             closingBinder(binder).registerExecutor(Key.get(ScheduledExecutorService.class, ForSegmentPruner.class));
         }
+
+        install(conditionalModule(
+                FileSystemSpoolingConfig.class,
+                fileSystemConfig -> fileSystemConfig.getLayout() == SIMPLE,
+                layoutBinder -> layoutBinder.bind(FileSystemLayout.class).to(SimpleFileSystemLayout.class)));
+
+        install(conditionalModule(
+                FileSystemSpoolingConfig.class,
+                fileSystemConfig -> fileSystemConfig.getLayout() == PARTITIONED,
+                layoutBinder -> {
+                    configBinder(layoutBinder).bindConfig(PartitionedLayoutConfig.class);
+                    layoutBinder.bind(FileSystemLayout.class).to(PartitionedFileSystemLayout.class);
+                }));
```

Comment:
Like I said above - I think this should be based on FS type

Reply:
I disagree - for smaller deployments, you probably don't care about API quotas and the default (SIMPLE) is enough. For larger ones - you want to configure the amount of partitions